### ISSUE TITLE
tile groups don't "stick"

### ISSUE THREAD

**Issue by fallenguru** [Leleat/Tiling-Assistant#284]
Tile groups get broken up all the time, i.e. the windows behave as individual windows again. I haven't been able to pinpoint a cause, but it looks like it happens every time my monitor goes to sleep / wakes up.

**System Info:**
 - Distro (incl. version): Ubuntu 22.04.3
 - GNOME Shell version: 42.9-0ubuntu2
 - Extension version and from where (e. g. EGO, `main` branch...): 36 [according to `metadata.json`] from EGO
 - XOrg/Wayland: XOrg
 - GPU (and driver): AMD Radeon VII (mesa 23.1.6\~kisak1\~j)

**Comment by Leleat** [comment]
>  every time my monitor goes to sleep / wakes up.

Is it just the monitor going to sleep is or are your locking your session? If so, could this be the same as https://github.com/Leleat/Tiling-Assistant/issues/258?

**Comment by Leleat** [comment]
> That said, according to metadata.json I have version 36 of Tiling Assistant? I didn't realise EGO was so far behind. Looks like my version of gnome-shell is not supported by recent versions of TA any more, even though it's the one in the current Ubuntu LTS ... :-( Any chance of this being backported?

I'll usually only support the latest GNOME Shell nowadays since ... well, it's just easier on my side (because I am on Fedora). Backporting the extension didn't use to be a thing because EGO only supports 1 version.  I heard EGO gained better support for multi-versioning somewhat recently, I'll need to look into it. 

But in the meantime you can just edit the following lines to fix the issue for yourself:

```js
const openWindows = this._twm.getWindows(false);
```
to
```js
const openWindows = this._twm.getWindows(true);
```
in lines 184 and 231 of  `~/.local/share/gnome-shell/extensions/tiling-assistant@leleat-on-github/extension.js`. After editing the file, you need to restart your GNOME Shell (e. g. by logging out).

**Comment by fallenguru** [comment]
> But in the meantime you can just edit the following lines to fix the issue for yourself: [...]

Just in case anyone else comes across this: You can attach `.patch` to GitHub commit URLs, like so: https://github.com/Leleat/Tiling-Assistant/commit/d44b8b6639267e541365665c5d756a23ff89c1d4.patch [that's the fix for #258]. Gives you a neat patch file to use with, well, `patch`. Applies cleanly, too.

Thanks! Will test for a couple of days and report back / close this.

**Comment by fallenguru** [comment]
My tiling groups stick like superglue now, thank you!
